snapcraft/_store.py

```diff
@@ -409,11 +409,14 @@ def push(snap_filename, release_channels=None):
         raise FileNotFoundError(
             'The file {!r} does not exist.'.format(snap_filename))
 
-    logger.info('Uploading {}.'.format(snap_filename))
-
     snap_yaml = _get_data_from_snap_file(snap_filename)
     snap_name = snap_yaml['name']
     store = storeapi.StoreClient()
+
+    with _requires_login():
+        store.push_precheck(snap_name)
```

Comment:
Does this step take any significant amount of time? Should a message be printed about what it's doing in case it hangs at all?